### ISSUE TITLE
Using with objects and arrays

### ISSUE THREAD

**Issue by bickmaev5** [vinissimus/next-translate#157]
Hello everyone

I’m switching from next-i18next to this lib and I had a problem using translation with arrays and objects in locales like here https://www.i18next.com/translation-function/objects-and-arrays

For example, i have this on locales json file
![image](https://user-images.githubusercontent.com/13235737/81944979-ee51b300-9616-11ea-90e3-242ff9370789.png);

and in component i use
`t(items_1, { returnObjects: true)`

and now i can use it like array
![image](https://user-images.githubusercontent.com/13235737/81945740-ea726080-9617-11ea-87a4-5e9faf61fff0.png)

is there a solution with objects and arrays in locales here?

**Comment by bickmaev5** [comment]
That work fine, but maybe lib have a native way?
src/I18nProvider.js
```
....
function getDicValue(dic, key = '', options = { returnObjects: false}) {
  const value = key.split('.').reduce((val, key) => val[key] || {}, dic)

  if (typeof value === 'string' || (value instanceof Object && options.returnObjects)) {
    return value;
  }
}
....
export default function I18nProvider({
  lang,
  namespaces = {},
  children,
  internals = {},
}) {
  const ns = useContext(NsContext)
  const allNamespaces = { ...ns, ...namespaces }

  setInternals({ ...internals, lang })

  function t(key = '', query, options) {
    const k = Array.isArray(key) ? key[0] : key
    const [namespace, i18nKey] = k.split(':')
    const dic = allNamespaces[namespace] || {}
    const keyWithPlural = plural(dic, i18nKey, query)
    const value = getDicValue(dic, keyWithPlural, options)

    return interpolation(value, query) || k
  }

  return (
    <I18nContext.Provider value={{ lang, t }}>
      <NsContext.Provider value={allNamespaces}>{children}</NsContext.Provider>
    </I18nContext.Provider>
  )
}

```

**Comment by aralroca** [comment]
Sorry, this is not supported. However, it can be interesting to implement 😊

Feel free to open a PR with your solution. Is your solution working well with params?

**Comment by bickmaev5** [comment]
Hi, @aralroca 
I have not tried it on the project.
Tested through jest and tests passed (at least locally :) )

opened PR https://github.com/vinissimus/next-translate/pull/158

**Comment by aralroca** [comment]
@bickmaev5 I merged the PR. Thank you for your contribution! 👏 It will be available on `0.16.0`, currently in the pre-release [0.16.0-canary.8](https://github.com/vinissimus/next-translate/releases/tag/0.16.0-canary.8).
